dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageView.java

```diff
@@ -124,13 +130,27 @@ public Template getTemplate() {
 
         containers.stream().forEach(containerRaw -> {
 
-            if (containerRaw.getContainer() instanceof FileAssetContainer) {
+            final Container container = containerRaw.getContainer();
 
-                final String path = FileAssetContainer.class.cast(containerRaw.getContainer()).getPath();
-                containerRawMap.put(path, containerRaw);
+            if (container instanceof FileAssetContainer) {
+                try {
+                    final Host host = APILocator.getHostAPI().findParentHost(container, APILocator.systemUser(), false);
```

Comment:
Instead of doing this, we should store the host id into the FileAssetContainer and compare with the current host

Reply:
done
https://github.com/dotCMS/core/pull/18033/commits/286b9c09f52d6e0bb78ec9761ec3000c06d7f38a#diff-96b7f9fe4f01ac03ce5bd6c461aa5d88R137

https://github.com/dotCMS/core/pull/18033/commits/286b9c09f52d6e0bb78ec9761ec3000c06d7f38a#diff-1854786c11c9b31c822e5b175c0f47c4R355